packages/app-mobile/components/screens/ExportScreen/NoteExportComponent.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react';
+import { Text, Alert } from 'react-native';
+import { Dispatch } from 'redux';
+import { _ } from '@joplin/lib/locale';
+import Logger from '@joplin/lib/Logger';
+import { Button } from 'react-native-paper';
+import { useCallback, useState } from 'react';
+import shim from '@joplin/lib/shim';
+import { join } from 'path';
+import Share from 'react-native-share';
+import exportAllFolders, { makeExportCacheDirectory } from './utils/exportAllFolders';
+import { ExportScreenStyles } from './useStyles';
+
+const logger = Logger.create('NoteExportComponent');
+
+interface Props {
+	themeId: number;
+	styles: ExportScreenStyles;
+	dispatch: Dispatch;
+}
+
+enum ExportStatus {
+	NotStarted,
+	Exporting,
+	Exported,
+}
+
+export const NoteExportComponent = (props: Props) => {
+	const [exportStatus, setExportStatus] = useState<ExportStatus>(ExportStatus.NotStarted);
+
+
+	const startExport = useCallback(async () => {
+		setExportStatus(ExportStatus.Exporting);
+		const exportTargetPath = join(await makeExportCacheDirectory(), 'jex-export.jex');
+		logger.info(`Exporting all folders to path ${exportTargetPath}`);
+
+		try {
+			await exportAllFolders(exportTargetPath);
+			// TODO: Use exportResult.warnings
+
+			await Share.open({
+				type: 'application/jex',
+				filename: 'export.jex',
+				url: `file://${exportTargetPath}`,
+				failOnCancel: false,
+			});
+
+			setExportStatus(ExportStatus.Exported);
+		} catch (e) {
+			logger.error('Unable to export:', e);
+
+			// Display a message to the user (e.g. in the case where the user is out of disk space).
+			Alert.alert(_('Error'), _('Unable to export or share data. Reason: %s', e.toString()));
+			setExportStatus(ExportStatus.NotStarted);
+		} finally {
+			await shim.fsDriver().remove(exportTargetPath);
+		}
+	}, []);
+
+	const cancelExport = useCallback(async () => {
+		// TODO: Implement
+	}, []);
```

Comment:
I'm not sure how long export will take on mobile for large note databases.

Implementing this will require refactoring of `InteropService`, and thus may be best left to a separate pull request.

Reply:
For my full notes database, a JEX export takes about 15 minutes on an Android tablet.